lib/rouge/lexers/robot.rb

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*- #
+module Rouge
+  module Lexers
+    class Robot < RegexLexer
+      tag 'robot'
+      aliases 'robotframework'
+
+      title "Robot"
+      desc 'Test DSL ( http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html )'
+
+      filenames '*.robot'
+      mimetypes 'text/x-robot'
+
+      def self.analyze_text(text)
+        return 1 if text =~ /\*\*\*\s+Setting(s)?\s+\*\*\*/
+      end
+
+      SETTINGS = '(Resource)|(Variables)|(Documentation)|(Metadata)|(Suite Setup)|(Suite Teardown)|(Force Tags)|(Default Tags)|(Test Setup)|(Test Teardown)|(Test Template)|(Test Timeout)|(Library)'
+      BDD = 'Given|When|Then|And|But'
+      TEST_CASE = 'Documentation|Tags|Setup|Teardown|Template|Timeout'
+      KEYWORD = 'Documentation|Tags|Arguments|Return|Teardown|Timeout'
+      rest_of_line = /.*?(?=[#\n])/
+
+
+      state :basic do
+#        #In robot framework tabs are very important, 
+#        #but highliting with a '→' sign can be an isssue for copypaste
+#        rule /\t/ do |m|
+#         #token Punctuation, '→' + m[0] 
+#        end
```

Comment:
(Also, it's a strong guarantee that rouge lexers output the same text that they are given, if you sum up all the tokens.)